### PR TITLE
Update link documentation in User Guide and address link feature bugs

### PR DESCRIPTION
**Pull request by shwene**
This PR addresses the two issues #137 and #141 with changes to regex and feedback message. It also improves on the user guide (links). 
Opening of links from Plannit has been refactored such that it will be opened via Link class instead of 2 separate but duplicate methods in OpenLinkCommand.java and ModuleCard.java 

### Issue #141
It raises the functionality bug of the add-link feature whereby valid links are being incorrectly rejected. After several rounds of testing, a few more bugs related to link acceptance were uncovered.

1. Valid links with subdomains other than (www.) was being rejected. (e.g. 'open.kattis.com', and 'teams.microsoft.com')
2. Valid links with top level domains longer than 6 characters was being rejected (though less common, there are quite a few examples listed here: https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains#ICANN)
3. Invalid links with symbols such as @ and \ within its domain name was being accepted. (e.g. email addresses)

##### Subdomain, Domain, and Top-Level Domain 
Only alphanumeric and hyphen characters can be included
Source:  www.rfc-editor.org/rfc/rfc3986 (only applies to English links, as there has been an update to domain names where other languages are valid also)

##### Remaining part of the URL (i.e. path/queries)
Printable ASCII characters (besides whitespace) are allowed. 
Rationale: Although there are conventions, organizations are free to decide on what characters to use. (Unlike the domain parts which are regulated).
Limitation: Non-ASCII characters (e.g. foreign language characters) are not considered due to parsing limitations of Plannit/Java/Json to store and open such characters.

### Issue #137
It raises the functionality bug add-link feature whereby the feedback message does not match the validation check being performed. As such, it has been updated to match the new link validation requirements.

### User Guide improvements

- Added user-centric explanation of and motivation behind link aliases (with example)
- Update constraints for link URLs in the relevant tables
- Included before and after images of add/delete link (excluded for open link as it is browser dependent)
- Reformatted table to ensure consistency of table formatting (as per @ekweirui 's suggestion)
- Updated command format explanation for `*` syntax (as per @cheeheng 's suggestion)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -499,44 +464,33 @@ delete-link m/CS2040 la/nus mods la/kattis
...
+
+| Before executing the command                                                       | After executing the command                                                       |
+|------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------|
+| {::nomarkdown}<p align="center"><img src="images/delete-link-before.png"/></p>{:/} | {::nomarkdown}<p align="center"><img src="images/delete-link-after.png"/></p>{:/} |
 
 #### 2.3.3. Open link
 You may open link(s) from a specific module to your default browser using the `open-link` command.
 Links will be opened by means of their corresponding alias.
 
+| :bulb: **Tip:** Hover your cursor over aliases in Plannit to peak at their associated URL |
```
Comment (teoyuqi) [inline docs/UserGuide.md:477]: I believe you have to use the HTML format for the infoboxes in order for them to be displayed correctly:
```
<div markdown="span" class="alert alert-info"> :bulb: **Tip:**<br/>  
Hover your cursor over aliases in Plannit to peak at their associated URL.
</div>
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -499,44 +464,33 @@ delete-link m/CS2040 la/nus mods la/kattis
...
-  <td><code>la/</code></td>
-  <td>
-    <li>Alphanumeric and whitespace characters</li>
-    <li>Trailing and leading whitespace characters will be removed</li>
-    <li>Character limit of 15</li>
-    <li>At least 1 alphanumeric character</li>
-  </td>
-</tr>
-</table>
+| :exclamation: **Alert:** Permissions from your operating system may be required for some users to open links from Plannit using either way<br>(They are enabled by default for most users) |
```
Comment (teoyuqi) [inline docs/UserGuide.md:486]: _see above comment_

**Review by teoyuqi — CHANGES_REQUESTED** [review]
Some minor changes. Do remember to update screenshots with labels as well.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -382,53 +382,32 @@ home page and view more details of a specific module, you may use the
...
-All links in Plannit are composed of a URL and a user-defined alias (case-sensitive).
+All links in Plannit are composed of a URL and an alias (case-sensitive). Aliases are short nicknames defined by you 
+for each module-related link that you found to be helpful. The succinct nature of aliases empowers you to identify 
+and launch module-related links from Plannit in a swift manner. Additionally, the personalised nature of aliases 
+enables you to attach your own meanings to generically-named links, saving you the
+hassle of searching and remembering links.
 
-| :exclamation: Plannit provides no guarantee of the link URL's existence |
-|-------------------------------------------------------------------------|
+E.g., Using the alias `tut t10 v fun` to label the link `nus-sg.zoom.us/j/81374731875?pwd=ZE9hWFlNVGJVNENFQm1SREhVVEs5Zz09`
```
[the remark this answers is not in the document]
Comment (teoyuqi) [reply docs/UserGuide.md:391]: Maybe `Tutorial Zoom`?

**Review by cheeheng — CHANGES_REQUESTED** [review]
It looks like there is an important security issue here, where the password of a Zoom session is being leaked to the public.

File: docs/UserGuide.md (hunk 1 above)
Comment (shwene) [reply docs/UserGuide.md]: Thanks! Will apply to the others too

File: docs/UserGuide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (shwene) [reply docs/UserGuide.md]: Ah, I wanted to showcase how aliases could be personalised

**Review by cheeheng — APPROVED** [review]
LGTM

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM

**Review by teoyuqi — APPROVED** [review]

**Pull request merged**